fake-koji/src/main/java/org/fakekoji/jobmanager/project/JDKProjectParser.java

```diff
@@ -381,4 +390,303 @@ void setProjectVariables(List<OToolVariable> variables) {
         }
     }
 
+    private static <T> Optional<T> findConfig(Set<T> configs, Predicate<T> predicate) {
+        return configs.stream()
+                .filter(predicate)
+                .findFirst();
+    }
+
+    private static Optional<PlatformConfig> findPlatform(Set<PlatformConfig> configs, String id, String provider) {
+        return findConfig(configs, config -> config.getId().equals(id) && config.getProvider().equals(provider));
+    }
+
+    private static PlatformConfig findOrCreatePlatform(Set<PlatformConfig> configs, String id, String provider) {
```

Comment:
or create??? It sounds bad....

Reply:
maybe a little :) but as it's creating a tree structure from jobs, some tree paths are shared between jobs...I didn't come up with better name :D

Reply:
As this is part of (in future) extraced code to new class, it very likely deserves some javadoc. it is very hard to read logic